syntax = "proto3";

package insider.sdk;

enum TransProto {
    ARP = 0;
    TCP = 1;
    UDP = 2;
}

enum DataProto {
    DATA_UPGRADE = 0;   // 升级数据
    DATA_PARA_OPERATION = 1;  // 参数操作
    DATA_ALARM = 2;     // 告警
    DATA_HEART_BEAT = 3;    // 心跳
    DATA_AI = 4;    // 深度学习任务
    DATA_PICTURE = 5;   // 图像
}

message Host {
    int32 port = 1;
    string ipv4 = 2;
}

enum DataFlow {
    DATA_IN = 0;  // 输入数据
    DATA_OUT = 1;  // 输出数据
}

message DataTime {
    int64 in_time = 1;
    int64 out_time = 2;
}

message Header {
    int64 msg_magic = 1;
    int64 pack_len = 2;
    int64 pack_id = 3;
    DataTime time = 4;
    DataFlow data_dir = 5;
    Host host = 6;
    TransProto trans_proto = 7;
    DataProto data_proto = 8;
}